# STAT 301: Midterm 1

## Data Wrangling



In [2]:
# The code will not be displayed in the quiz
library(tidyverse, quietly = T)
library(broom, quietly = T)
library(repr, quietly = T)
library(infer, quietly = T)
library(ggplot2)
library(cowplot)
library(contextual)

dat <- read.csv("Assessment_2015.csv")
dat <- dat %>% filter(ASSESSCLAS=="Residential")  %>% filter(ASSESSMENT < 1500000)

dat <- dat  %>% mutate(BLDG_AGE = cut(YEAR_BUILT,
                     breaks=c(-Inf,1980,2000,+Inf),
                     labels=c("plus40","mid40-20","new20")))

set.seed(123)
dat_tax <- rep_sample_n(dat, size = 1000)


## Question 1

Answer TRUE/FALSE to the following statement:

Sir Francis Galton hypothesized that a son’s height could be predicted using the father’s height and collected data to estimate a simple linear regression. In this model, the input (or explanatory) variable was the height of the father.

### BEGIN SOLUTION

TRUE

### END SOLUTION

## Question 2

Answer TRUE/FALSE to the following statement:

An *additive* multiple linear regression with one continuous input variable and one categorical input variable with four categories has 4 *slopes* to be estimated.

### BEGIN SOLUTION

FALSE, the additive model has a unique slope shared by all the levels of the categorical variable. We assume that the association between the response and the continuous covariate is the *same* for all levels of the categorical variable.

### END SOLUTION


## Question 3

Answer TRUE/FALSE to the following statement:

Using bootstrapping, we can approximate the standard error of the estimator of the slope $\hat{\beta}_1$ by the standard deviation of the bootstrap estimates.

### BEGIN SOLUTION

TRUE

### END SOLUTION

## Question 4

Answer TRUE/FALSE to the following statement:

Using data, we compute a 95% confidence interval for the population slope $\beta_1$ of a simple linear regression and obtain the interval (30.3, 31.6). Based on this result we can claim that the population slope is between 30.3 and 31.6 with a 95% probability. 


### BEGIN SOLUTION

FALSE, it should say "95% confidence". Once the CI is built using data, there is no more randomness to measure a *probability*. If CIs are computed based on *different* samples, we are confident that 95% of them will contain the true slope. The one based on your data may or may not cover the true slope.

### END SOLUTION

## Question 5

The `tidy()` function displays the column `std.error`. List one possible use or interpretation for these values in this column in one or two sentences.

### BEGIN SOLUTION

- The `std.error` column can be used to estimate the margin of errors in the confidence intervals for each one of the model estimates.
- Moreover, each value in this column can be used to compute the test statistic $t$-value to be used in hypothesis testing to compute the corresponding $p$-value associated with it under the null hypothesis $H_0$.
- We can also use these values to measure an estimator's expected variation due to random sampling.

**Notes on marking:**

- One use is enough to give the full mark.

### END SOLUTION



## Question 6

We want to investigate which variables affect the tax assessed values of properties. We are going to use a dataset called `dat_tax`, that contains valuations and property conditions of 1000 properties randomly selected from data collected in 2014 for all residencial properties in the Strathcona County, Alberta. 

Among other variables in the dataset, we will focus on the following 3 variables:

- `ASSESSMENT`: the assessed value of the property (in $)


- `BLDG_METRE`: the size of the property (in mts)
        

- `BLDG_AGE`: the age of the property as a categorical variable with 3 levels: "plus40" (selected as the reference level for properties built more than 40 years ago) , "mid40-20" (for properties that are 40 to 20 years old) and "new20" (for properties that were built less than 20 years ago).

**Question 6.1**

Provide the R code to estimate a simple linear regression to model the relation between the response and a continuous input variable. Use the name of the variables given above.

### BEGIN SOLUTION

> lm_size<- lm(ASSESSMENT ~ BLDG_METRE, data = dat_tax)

### END SOLUTION

**Question 6.2**

Using `dat_tax`, we estimate a simple linear regression using `lm()` to study the relation between the assessed value of a property and its size. Results are shown in the table below obtained with `tidy()`. In one sentence, interpret the estimate of the parameter for `BLDG_METRE`. 


In [2]:
# The code will not be displayed in the quiz
lm_size<- lm(ASSESSMENT ~ BLDG_METRE, data = dat_tax)
tidy(lm_size)  %>% mutate_if(is.numeric, round, 3)

term        estimate   std.error statistic p.value
1 (Intercept) 127317.865 10033.463 12.689    0      
2 BLDG_METRE    2356.108    62.966 37.419    0

### BEGIN SOLUTION

An additional meter in the size of the property is associated with an expected increase in the assessed value of $2356.108.

### END SOLUTION

## Question 7

Using `dat_tax`, we want to investigate if the assessed value of a property is associated with its age measured by `BLDG_AGE`. We fit a linear regression model with `lm()`. The output of `tidy()` is shown below.

In [3]:
# The code will not be displayed in the quiz

lm_size_age_add <- lm(ASSESSMENT ~ BLDG_METRE + BLDG_AGE, data = dat_tax)
tidy(lm_size_age_add)  %>% mutate_if(is.numeric, round, 3)

term             estimate   std.error statistic p.value
1 (Intercept)      125789.449 9916.236  12.685    0      
2 BLDG_METRE         2214.924   66.918  33.099    0      
3 BLDG_AGEmid40-20  35158.027 7953.204   4.421    0      
4 BLDG_AGEnew20     43237.149 8369.910   5.166    0

**Question 7.1**

Based on the estimates and results in the table given by `tidy()`, determine if the model contains any interaction term and in one or two sentences justify your answer.

### BEGIN SOLUTION

The model does not contain an interaction term. There are only 4 parameters to characterize 3 lines with equal slopes. 

The model does not contain an interaction term. There is only one common slope given by the estimate of `BLDG_METRE`.

### END SOLUTION

**Question 7.2**

In one or two sentences, explain which assumption was made about the relation of the variables to propose such a model (i.e., with or without interactions). In other words, what was assumed about the relation between the assessed value and the size for properties of different ages?

### BEGIN SOLUTION

An additive model was fit since it was assumed that the association between the assessed value and the size of the property is the same for properties of all ages. The slope is the same for all levels of `BLDG_AGE`. 

### END SOLUTION

**Question 7.3**

Which of the following plots matches the model fit and summarized in the previous table? In one or two sentences, justify your choice and explain how the estimates given in the table relate with the line(s) in your selected plot.

### BEGIN SOLUTION

PLOT B: An additive model is represented by 3 paralell lines slope. The common slope is equal to 2214.924. The intercept of the reference red line is 125789.449. The intercept of the green line is (2214.924 + 35158.027). The intercept of the blue line is (2214.924 + 43237.149).

### END SOLUTION

In [7]:
# The code will not be displayed in the quiz
options(repr.plot.width = 15, repr.plot.height = 7)

#SLR
dat_tax$pred_slm <- predict(lm_size)

lm_size_plot <- ggplot(dat_tax, aes(
  x = BLDG_METRE,
  y = ASSESSMENT,
  color = BLDG_AGE
)) +
  geom_point() +
  geom_line(aes(y = pred_slm), size = 1, color = "black") +
  labs(
    title = "PLOT A",
    x = "Building Size (mts)",
    y = "Assessed Value ($)"
  ) +
  theme(
    text = element_text(size = 16.5),
    plot.title = element_text(face = "bold"),
    axis.title = element_text(face = "bold"),
    legend.title = element_text(face = "bold"),
  ) +
  labs(color = "Building Age")

#Additive
dat_tax$pred_lm_add <- predict(lm_size_age_add)

lm_size_age_add_plot <- ggplot(dat_tax, aes(
  x = BLDG_METRE,
  y = ASSESSMENT,
  color = BLDG_AGE
)) +
  geom_point() +
  geom_line(aes(y = pred_lm_add), size = 1) +
  labs(
    title = "PLOT B",
    x = "Building Size (mts)",
    y = "Assessed Value ($)"
  ) +
  theme(
    text = element_text(size = 16.5),
    plot.title = element_text(face = "bold"),
    axis.title = element_text(face = "bold"),
    legend.title = element_text(face = "bold"),
  ) +
  labs(color = "Building Age")

#Interaction
lm_size_age_int <- lm(ASSESSMENT ~ BLDG_METRE * BLDG_AGE, data = dat_tax)
dat_tax$pred_lm_int <- predict(lm_size_age_int)
png(filename="faithful.png")
lm_size_age_int_plot <- ggplot(dat_tax, aes(
  x = BLDG_METRE,
  y = ASSESSMENT,
  color = BLDG_AGE
)) +
  geom_point() +
  geom_line(aes(y = pred_lm_int), size = 1) +
  labs(
    title = "PLOT C",
    x = "Building Size (mts)",
    y = "Assessed Value ($)"
  ) +
  theme(
    text = element_text(size = 16.5),
    plot.title = element_text(face = "bold"),
    axis.title = element_text(face = "bold"),
    legend.title = element_text(face = "bold"),
  ) +
  labs(color = "Building Age")

png(filename="/home/lourenzutti/question10.png", width = 1000, height = 750)
plot_grid(lm_size_plot, lm_size_age_add_plot, lm_size_age_int_plot)
dev.off()

png 
  2

## Question 8

Using `dat_tax`, and fitting a model *with interactions*, we obtain an estimate for the term `BLDG_METRE` together with results of a test of hypothesis and 95% confidence intervals. Based on the output displayed below, select all the claims that are *correct*:

In [5]:
# The code will not be displayed in the quiz
tidy(lm_size_age_int, conf.int = TRUE)  %>% mutate_if(is.numeric, round, 3) %>% 
    subset(term == "BLDG_METRE")


term       estimate std.error statistic p.value conf.low conf.high
1 BLDG_METRE 1768.566 120.009   14.737    0       1533.065 2004.067

**A.** at significance level of 5%, the assessed value of a property more than 40 years old is statistically different from that of houses less than 20 years old.

**B.** at significance level of 5%, the assessed value of any property is significantly associated with the size of the property.

**C.** at significance level of 5%, we have enough evidence to reject the null hypothesis that the assessed value of any property are not associated with the size of the property.

**D.** at significance level of 5%, we have enough evidence to reject the null hypothesis that the  assessed value of properties more than 40 years old are not associated with the size of the property.

**E.** with 95% confidence, we expect that the assessed value of a property increases between \\$1533.065 and \\$2004.067 for every 1 additional metre in size.

**F.** with 95% confidence, we expect that the assessed value of a property increases between \\$1533.065 and \\$2004.067.

**G.** with 95% confidence, the expected assessed value of a property more than 40 years old is between \\$1533.065 and \\$2004.067.

### BEGIN SOLUTION

D is the only correct one. All other options do not state "for properties with more than 40 years" (reference value) or refer to a wrong parameter in a model *with* interaction terms.

### END SOLUTION